storage/backend.go

```diff
@@ -323,6 +324,35 @@ func (b *Backend) RemoveVolume(vol *Volume) error {
 	return nil
 }
 
+func (b *Backend) CreateVolumeSnapshot(snapshotName string, volConfig *VolumeConfig) (*Snapshot, error) {
+
+	log.WithFields(log.Fields{
+		"backend":       b.Name,
+		"storage_class": volConfig.StorageClass,
+		"source_volume": volConfig.Name,
+		"snapshot_name": snapshotName,
+	}).Debug("Attempting snapshot create.")
+
+	// Prepare volume
+	if volConfig.InternalName == "" {
```

Comment:
You shouldn't need this check.  After it is created, a volume always has an internal name, which is its name on the storage controller.  (The volume name is always the PV name.  The internal name may closely resemble the PV name, or it may differ completely, depending on the needs of the backing storage.)